src/coefficient_structs.jl

```diff
@@ -0,0 +1,33 @@
+abstract type Coefficients end
+
+"""Coefficients of the generalised logistic function to describe the vertical coordinate.
+Default coefficients A,K,C,Q,B,M,ν are fitted to the old L31 configuration at ECMWF.
+See geometry.jl and function vertical_coordinate for more informaiton.
+
+Following the notation of https://en.wikipedia.org/wiki/Generalised_logistic_function (Dec 15 2021).
+
+Change default parameters for more/fewer levels in the stratosphere vs troposphere vs boundary layer."""
+@with_kw struct GenLogisticCoefs{NF<:Real} <: Coefficients
+    A::NF = -0.283     # obtained from a fit in /input_date/vertical_coordinate/vertical_resolution.ipynb
+    K::NF = 0.871
+    C::NF = 0.414
+    Q::NF = 6.695
+    B::NF = 10.336
+    M::NF = 0.602
+    ν::NF = 5.812
+end
+
+"""
+Parameters for computing saturation specific humidity.
+
+These parameters, and the formula where they are used, do not appear in the original
+Speedy documentation.
+"""
+@with_kw struct HumidityCoefs{NF<:Real} <: Coefficients
+    e₀::NF = 6.108e-3
+    C₁::NF = 17.269
+    C₂::NF = 21.875
+    T₀::NF = 273.16
+    T₁::NF = 35.86
+    T₂::NF = 7.66
```

Comment:
Do any of these have any physical meaning? I guess `T₀` (also maybe we should add that to the style guide, is there an advantage in calling things T₀ over T0?) is the conversion between ˚C and K?

Reply:
So these are the parameters for Magnus' formula for saturation vapour pressure, and as far as I can tell they're determined experimentally. So none of them have an obvious physical meaning, except for `T₀` of course, which in this case is actually used to convert from Celsius to Kelvin.

`C₁` and `T₁` are for the saturation vapour pressure over water, while `C₂` and `T₂` are over ice. Speedy uses the first set of parameters if the temperature is above zero, and the second set if it's below zero. From what I can tell this isn't strictly correct, since apparently water in clouds is often cooled below zero, but such an approximation is probably consistent with the approximate nature of Speedy's parameterisations?

There appears to be a long history of coming up with new, ever so slightly different values for these parameters. A lot of them are listed [here](https://www.osti.gov/servlets/purl/548871). Yet nowhere have I managed to find the exact values used in Speedy!

In any case, it seems like the modern accepted values can be found in Table 8.3, p. 211 of the [Springer Handbook of Atmospheric Measurements](https://link.springer.com/book/10.1007/978-3-030-52171-4) ;) (I would certainly keep the original values from Speedy, for now.)

I'll probably refactor this a bit, to have something like a `MagnusCoefficients` struct, with separate instances for water and ice. I think that's the most logically consistent approach, and also makes it easier to substitute more up-to-date values.

Reply:
Regarding `T₀` over `T0`, it's just a personal preference to have the code resemble the maths (I find it easier to spot mistakes that way). But happy to get rid of them subscripts :)

Reply:
I like `HumidityCoefficients` and the idea to access them for example via `M.constants.humidity_coefficients`. Not particularly hard feelings for `T₀` over `T0`, I'd just make sure that the lines that use them get a comment like `# T0 = zero ˚C in Kelvin`